starknet_py/net/account/account.py

```diff
@@ -124,18 +122,12 @@ async def _get_max_fee(
 
         return max_fee
 
-    async def _prepare_invoke(
-        self,
-        calls: Calls,
-        max_fee: Optional[int] = None,
-        auto_estimate: bool = False,
-    ) -> Invoke:
+    async def prepare_invoke(self, calls: Calls, max_fee: int = 0) -> Invoke:
```

Comment:
I'm still unsure about this method. It is a bit weird having method like that which is useful mostly as a helper for the other account methods exposed as public. 

If you think there is a valid reason to expose this I think it should be made into static/completely separate method.

Reply:
cc @war-in

Reply:
Agreed, why was it exposed?

Reply:
It's only used in `Account.sign_invoke_transaction` as a 'helper'. It's an independent method, that has its own use cases.

I don't really see why it shouldn't be a part of Account api. The method creates an unsigned invoke transaction from calls. There is no other exposed function in the library to create Invoke from calls apart from `sign_invoke_transaction`, which even from the name does not suggest it does that - it suggests it signs an Invoke transaction.

Also, if it wasn't a part of Account, the version, nonce, contract_address essential fields would need to be filled manually later, after using this method, for the transaction to be sent (its fee to be estimated).

Reply:
So we could either leave `prepare_invoke` exposed as it is, or not support the `skip_validate` flag.
I don't see any other solution that would make sense here.
@war-in